cmd/check_spec.go

```diff
@@ -122,58 +92,419 @@ func newCheckSpec(ctx context.Context) (*checkSpec, error) {
+	if fetchErr != nil {
+		fmt.Printf("%v: unable to get network status\n", fetchErr.Err)
+		setValidationStatus(output, defaultTip, checkSpecFailure)
+		return output
+	}
+	tip = res.CurrentBlockIdentifier
+	if tip.Index-block.BlockIdentifier.Index > tipDiff {
+		fmt.Printf("%v\n", errBlockTip)
+		setValidationStatus(output, defaultTip, checkSpecFailure)
 	}
 
-	// TODO: more checks
+	fmt.Println("/block validated")
+	return output
+}
+
+func (cs *checkSpec) errorObject(ctx context.Context) checkSpecOutput {
+	output := checkSpecOutput{
+		api: errorObject,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			errorCode:    checkSpecSuccess,
+			errorMessage: checkSpecSuccess,
+		},
+	}
+
+	// Send an empty request to /network/status and verify error response
+	emptyNetwork := &types.NetworkIdentifier{}
+	_, err := cs.onlineFetcher.NetworkStatusRetry(ctx, emptyNetwork, nil)
+	if err.ClientErr.Code < 0 {
+		fmt.Printf("%v\n", errErrorNegativeCode)
+		setValidationStatus(output, errorCode, checkSpecFailure)
+	}
+
+	if err.ClientErr.Message == "" {
+		fmt.Printf("%v\n", errErrorEmptyMessage)
+		setValidationStatus(output, errorMessage, checkSpecFailure)
+	}
+
+	fmt.Println("error object validated")
+	return output
+}
+
+// Searching for an account backwards from the tip
+func (cs *checkSpec) getAccount(ctx context.Context) (
+	*types.AccountIdentifier,
+	*types.PartialBlockIdentifier,
+	[]*types.Currency,
+	error) {
+	res, err := cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("%v: unable to get network status", err.Err)
+	}
+
+	var acct *types.AccountIdentifier
+	var blockID *types.PartialBlockIdentifier
+	tip := res.CurrentBlockIdentifier.Index
+	genesis := res.GenesisBlockIdentifier.Index
+	currencies := []*types.Currency{}
+
+	for i := tip; i >= genesis && acct == nil; i-- {
+		blockID = &types.PartialBlockIdentifier{
+			Index: &i,
+		}
+
+		block, err := cs.onlineFetcher.BlockRetry(ctx, Config.Network, blockID)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("%v: unable to fetch block at index: %v", err.Err, i)
+		}
+
+		// looking for an account in block transactions
+		for _, tx := range block.Transactions {
+			for _, op := range tx.Operations {
+				if op.Account != nil && op.Amount.Currency != nil {
+					acct = op.Account
+					currencies = append(currencies, op.Amount.Currency)
+					break
+				}
+			}
+
+			if acct != nil {
+				break
+			}
+		}
+	}
+
+	return acct, blockID, currencies, nil
+}
+
+func runCheckSpecCmd(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+	cs, err := newCheckSpec(ctx)
+	if err != nil {
+		return fmt.Errorf("%v: unable to create checkSpec object with online URL", err)
+	}
+
+	output := []checkSpecOutput{}
+	// validate api endpoints
+	output = append(output, cs.networkStatus(ctx))
+	output = append(output, cs.networkList(ctx, cs.offlineFetcher))
```

Comment:
will we check online fetcher as well?

Reply:
IMO it's not quite necessary, because based on our experience, AI tends to make mistakes with offline url, e.g make it equal to online url. The concern is to ensure the endpoint works in offline mode. If the answer is yes, then the same endpoint should work in online mode as well. But feel free to correct me.